lib/module.js

```diff
@@ -566,15 +566,20 @@ Module.prototype.load = function(filename) {
   if (ESMLoader) {
     const url = getURLFromFilePath(filename);
     const urlString = `${url}`;
+    const exports = this.exports;
     if (ESMLoader.moduleMap.has(urlString) !== true) {
-      const ctx = createDynamicModule(['default'], url);
-      ctx.reflect.exports.default.set(this.exports);
       ESMLoader.moduleMap.set(urlString,
-                              new ModuleJob(ESMLoader, url, async () => ctx));
+                              new ModuleJob(ESMLoader, url, async () => {
+          const ctx = createDynamicModule(
+            ['default'], url);
+          ctx.reflect.exports.default.set(exports);
+          return ctx;
+        })
+      );
```

Comment:
Nit: weird indentation

```js
      ESMLoader.moduleMap.set(
        urlString,
        new ModuleJob(ESMLoader, url, async () => {
          const ctx = createDynamicModule(
            ['default'], url);
          ctx.reflect.exports.default.set(exports);
          return ctx;
        })
      );
```

Also I suppose the `const exports = this.exports` isn't needed as it's used in an arrow function only anyway.

Reply:
need to save it due to timing difference.